### PR TITLE
Finance offer link

### PR DESCRIPTION
**Pull request by EvanBarbour3**
## Features
- Added `finance_offers` field, and new route so that applications can be viewed instead of started

[Delivers [#119172315](https://www.pivotaltracker.com/story/show/119172315)]

### CODE REVIEW

File: app/Basket/Installation.php [hunk 1]
```diff
@@ -58,6 +65,7 @@ class Installation extends Model
         'validity',
         'custom_logo_url',
         'disclosure',
+        'finance_offers'
```
Comment (wnowicki) [inline app/Basket/Installation.php]: CS `,`

File: composer.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
         "wnowicki/generic": "~0.0",
         "erusev/parsedown": "^1.5",
         "league/csv": "^7.1",
-        "paybreak/paybreak-sdk-php": "^4.0"
+        "paybreak/paybreak-sdk-php": "^4.0",
+        "paybreak/foundation": "~0.0"
```
Comment (wnowicki) [inline composer.json]: `~0.11` 😛

**Comment by EvanBarbour3** [comment]
@wnowicki Ready for your review

File: app/Basket/Installation.php [hunk 3]
```diff
@@ -34,12 +34,17 @@
...
+ * @property int    $finance_offers
  * @package App\Basket
  */
 class Installation extends Model
 {
     protected $table = 'installations';
 
+    const IN_STORE = 2;
+    const LINK = 4;
+    const LOWEST_BIT = self::IN_STORE;
```
Comment (wnowicki) [inline app/Basket/Installation.php]: ??

File: app/Basket/Installation.php [hunk 4]
```diff
@@ -145,4 +151,27 @@ public function getDisclosureAsHtml()
...
+     *
+     * @author EB
+     * @return array
+     */
+    public function getBitwiseFinanceOffers()
+    {
+        return [
+            'in_store' => [
+                'value' => self::IN_STORE,
+                'active' => $this->finance_offers&self::IN_STORE,
```
Comment (wnowicki) [inline app/Basket/Installation.php]: Should you use `Bitwise` here ?

File: app/Http/Controllers/InitialisationController.php [hunk 5]
```diff
@@ -118,6 +143,7 @@ public function chooseProduct($locationId, Request $request)
                 ),
                 'amount' => floor($request->get('amount') * 100),
                 'location' => $location,
+                'bitwise' => Bitwise::make($location->installation->finance_offers)
```
Comment (wnowicki) [inline app/Http/Controllers/InitialisationController.php]: - I'm not sure about passing objects to view

File: resources/views/initialise/main.blade.php [hunk 6]
```diff
@@ -251,25 +251,24 @@
...
-                                    </div>
-                                    @if($product['credit_info']['deposit_amount'] > 0)
-                                        <button type="submit" class="btn btn-success btn-lg btn-block">Continue with a &pound;{{ number_format($product['credit_info']['deposit_amount']/100, 2) }} deposit today
-                                            @if($product['credit_info']['amount_service'] > 0)
-                                                and a service fee of &pound;{{ number_format($product['credit_info']['amount_service']/100, 2) }}
+                                    @foreach($location->installation->getBitwiseFinanceOffers() as $key => $offer)
+
+                                        @if(count($bitwise->explode()) == 2)<div class="col-sm-6 col-xs-12">@endif
+
+                                            @if(in_array($offer['value'], $bitwise->explode()))
```
Comment (wnowicki) [inline resources/views/initialise/main.blade.php]: - `$bitwise->contains($offer['value'])`
- if you think about it `explode()` is quite heavy to put it in a loop

**Comment by wnowicki** [comment]
@EvanBarbour3

File: app/Http/Controllers/InitialisationController.php (hunk 5 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/InitialisationController.php]: `$location` is an object? We pass them in all the time??

**Pull request merged**